### PR TITLE
refactor: Make typer an optional dependency.

### PR DESCRIPTION
**Pull request by DanCardin**
Fixes https://github.com/gabrieldemarmiesse/python-on-whales/issues/512

### CODE REVIEW

**Comment by DanCardin** [comment]
Ideally I would add to this that `tqdm` is also part of the "cli" extra. perhaps also add a progress=False flag (which defaults to False if tqdm were not installed), but the CLI defaults to `True`. In my opinion, the vanilla calling of `python_on_whales` through code shouldn't be enabling tqdm progress bars by default (except through the CLI). As a library, this sort of output is very surprising.

However, I didn't include that so far given your comments in the original issue.

**Comment by gabrieldemarmiesse** [comment]
Thank you for the pull request. I didn't expect it to be that fast! I think that first we need to take into account backward compatibility. By merging this right now, everyone relying on the CLI will have their CI pipeline broken. We should warn them first, for a few releases, then set typer as an optional dependency. Suddenly adding a breaking change to the software is not something that we should do unless there is a security issue.

Let's leave this PR open, we can merge it in a few releases. If you're willing to add the warning, I would happily merge such a PR now.

**Comment by DanCardin** [comment]
The problem, as I see it, with warning; is that I dont see that there's anything the user can do to make the warning go away. 

I could add the extra, but leave the dependency as required. But as far as I know, there's no way to detect if a library was installed with an extra or not, so they'll get the warning even if they have switched to using the cli extra.

I would think use of the CLI would primarily be a local-use tool, rather than something that would break someone's build. but that's completely hearsay from my own personal use.

File: python_on_whales/command_line_entrypoint.py [hunk 1]
```diff
@@ -1,6 +1,11 @@
 from typing import Optional
 
-import typer
+try:
+    import typer
+except ImportError:
+    raise ImportError(
+        "The `python-on-whales` CLI, requires the 'cli' extra. `pip install python-on-whales[cli]`."
+    )
```
Comment (LewisGaul) [inline python_on_whales/command_line_entrypoint.py:8]: There should be a `from` here, either `from None` or `from exc` (making it `except ImportError as exc`).

**Comment by gabrieldemarmiesse** [comment]
I think this can be closed as typer is not a dependency anymore! 🎉

**Pull request closed without merging**